src/thunderbots/software/util/parameter/parameter.h

```diff
@@ -180,7 +224,9 @@ class Parameter
     {
         for (const auto& pair : Parameter<T>::getRegistry())
         {
-            pair.second->updateValueFromROSParameterServer();
+            pair.second.first->lock();
+            pair.second.second->updateValueFromROSParameterServer();
+            pair.second.first->unlock();
```

Comment:
Similar comment here about `scoped_lock`, and most other `lock`/`unlock` pairs

Reply:
fixed all lock/unlock to used scoped lock